### PR TITLE
refactor: detach request from context

### LINKED ISSUE #113: Change handler signature from `(ctx) => resp` to `(req, ctx) => resp`
The current signature is somewhat clumsy. It does not align nicely with the signature for `serve` from deno.land/std.

The `req` property from `ctx` should be removed. The `match` property from `ctx` should be renamed to `params`.

This is a prerequisite for #96.

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #113

### CODE REVIEW

File: src/server/types.ts [hunk 1]
```diff
@@ -33,30 +33,34 @@ export interface IslandModule {
 }
 
 export interface HandlerContext<T = unknown> {
-  req: Request;
   match: Record<string, string>;
```
Comment (lucacasonato) [inline src/server/types.ts:36]: Want to do this rename in a follow up?

**Review by lucacasonato — COMMENTED** [review]
www/ and examples/ and `fresh init` need to be updated.

File: src/server/types.ts (hunk 1 above)
Comment (crowlKats) [reply src/server/types.ts:36]: yea, was also considering doing #96 in this PR, but maybe its just cleaner to have these things in separate PRs

**Comment by lucacasonato** [comment]
Thx :-)

**Pull request merged**